### DISCUSSION TITLE
Avoid rerender on mousemove

### DISCUSSION THREAD

**Discussion by eduardfossas** [vercel/next.js#25255] [Help]
Hello, I know this is a react question but maybe with some Next magic it works.

I have a mousemove inside a functional component and each time I am moving the mouse it rerenders, is there a way of avoiding that in the provided example?

https://codesandbox.io/s/eager-cache-ygt8r

Kind regards and many thanks.

**Comment by Kab1r** [comment]
If the component doesn't re-render, the displayed value will not change.
You can use `useRef` instead of `useState` to achieve what you have described.
```jsx
import { useRef, useState } from "react";
import "./styles.css";

export const App = () => {
  const mouse = useRef({ x: 0, y: 0 });
  const [_, forceUpdate] = useState(0);
  const handleMouseMove = (e) => {
    mouse.current = {
      ...mouse.current,
      x: e.clientX,
      y: e.clientY
    };
  };

  console.log("render");

  return (
    <div
      className="App"
      onMouseMove={handleMouseMove}
      style={{ height: "100vh", width: "100vw" }}
    >
      <div>{(mouse.current.x, mouse.current.y)}</div>
      <button onClick={() => forceUpdate(!_)}>render</button>
    </div>
  );
};
```

**Reply by eduardfossas** [reply]
Thanks for the reply. I did as you said a few days ago and it worked. Thank you for your help
